.github/workflows/create-tag.yml

```diff
@@ -0,0 +1,71 @@
+name: "Create a tag"
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Branch to be tagged"
+        required: true
+      tag:
+        description: "Tag for new version (1.23.4)"
+        required: true
+
+jobs:
+  create-tag:
+    name: "Create a tag"
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Generate a token
+        id: generate_token
+        uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92
+        with:
+          app_id: ${{ secrets.TEMPORAL_CICD_APP_ID }}
+          private_key: ${{ secrets.TEMPORAL_CICD_PRIVATE_KEY }}
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: true
+          token: ${{ steps.generate_token.outputs.token }}
+          ref: ${{ github.event.inputs.branch }}
+          fetch-depth: 0
+
+      - name: Set up Github credentials
+        env:
+          CURRENT_REPO: ${{ github.repository }}
+          TOKEN: ${{ steps.generate_token.outputs.token }}
+        run: |
+          git config --local user.name 'Temporal Data'
+          git config --local user.email 'commander-data@temporal.io'
+          git remote set-url origin "https://x-access-token:$TOKEN@github.com/$CURRENT_REPO"
+
+      - name: Get current version
+        id: get_current_version
+        run: |
+          CURRENT_VERSION=$(grep '^\s*ServerVersion = ".*"$' common/headers/version_checker.go | sed 's/^.*"\(.*\)"$/\1/')
+          [ -z "$CURRENT_VERSION" ] && exit 1
+          echo "CURRENT_VERSION=$CURRENT_VERSION" >> "$GITHUB_OUTPUT"
+
+      - name: Update Server version
+        if: ${{ steps.get_current_version.outputs.CURRENT_VERSION != github.event.inputs.tag }}
+        env:
+          TAG: ${{ github.event.inputs.tag }}
+          BRANCH: ${{ github.event.inputs.branch }}
+        run: |
+          sed -i -e "s/ServerVersion = \".*\"$/ServerVersion = \"$TAG\"/g" common/headers/version_checker.go
```

Comment:
Should add a comment in that file to indicate that line is updated automatically by this script, so that no one get tempted to rename/move/whatever, and end up breaking the release process.